service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -347,6 +349,46 @@ protected final String getDeviceIdParam(final RoutingContext ctx) {
         return ctx.request().getParam(PARAM_DEVICE_ID);
     }
 
+    /**
+     * Get request parameter value and check if it has been set. If it's not set, fail the request.
+     *
+     * @param paramName The name of the parameter to get.
+     * @param ctx The routing context of the request.
+     * @param span The active OpenTracing span for this operation. In case of the missing mandatory parameter, the error is logged and the span is finished.
+     *             Otherwise, the parameter is set as a span tag.
+     * @return The value of the parameter if it's set or {@code null} otherwise.
+     * @throws NullPointerException If ctx is {@code null}.
+     */
+    protected final String getMandatoryRequestParam(final String paramName, final RoutingContext ctx, final Span span) {
+        return getRequestParam(paramName, ctx, span, false);
+    }
+
+    /**
+     * Get request parameter value. Optionally, if parameter has not been set, fail the request.
+     *
+     * @param paramName The name of the parameter to get.
+     * @param ctx The routing context of the request.
+     * @param span The active OpenTracing span for this operation. In case of the missing mandatory parameter, the error is logged and the span is finished.
+     *             Otherwise, the parameter is set as a span tag.
+     * @param optional Whether to check if parameter has been set or not.
+     * @return The value of the parameter if it's set or {@code null} otherwise.
+     * @throws NullPointerException If ctx is {@code null}.
```

Comment:
I am pretty sure that you will also get a `NullPointException` if the paramName is `null`.